crates/sui-core/src/consensus_adapter.rs

```diff
@@ -466,11 +467,16 @@ impl ConsensusAdapter {
                         .with_label_values(&[&bucket])
                         .observe(ack_start.elapsed().as_secs_f64());
                 });
-            }
-            debug!("Submitted {transaction_key:?} to consensus");
-            processed_waiter
-                .await
-                .expect("Storage error when waiting for consensus message processed");
+            };
+            match select(processed_waiter, submit_inner.boxed()).await {
+                Either::Left((processed, _submitted)) => {
+                    processed.expect("Storage error when waiting for consensus message processed");
+                    debug!("Processed {transaction_key:?} from consensus");
+                }
+                Either::Right(((), _processed_waiter)) => {
+                    debug!("Submitted {transaction_key:?} to consensus");
```

Comment:
We need to await for processed_waiter here to make sure we actually wait for transaction to appear in consensus when we submitted it

Reply:
Good point, fixed.